### ISSUE TITLE
LockedError: database table is locked

### ISSUE THREAD

**Issue by adamkrellenstein** [CounterpartyXCP/counterparty-core#1718]
```
32|counter | [2024-04-20 11:09:30][DEBUG] Performing light healthz check.
32|counter | [2024-04-20 11:09:30][INFO] Stopping API Server...
32|counter | [2024-04-20 11:09:30][INFO] Stopping backend...
32|counter | [2024-04-20 11:09:30][DEBUG] `localhost:8432` -- disconnected
32|counter | [2024-04-20 11:09:31][INFO] Running PRAGMA optimize...
32|counter | [2024-04-20 11:18:06][INFO] PRAGMA optimize done.
32|counter | [2024-04-20 11:18:06][INFO] Closing database...
32|counter | [2024-04-20 11:18:06][ERROR] Unhandled Exception
32|counter | Traceback (most recent call last):
32|counter |   File "/home/adam/counterparty-core/counterparty-core/counterpartycore/server.py", line 598, in start_all
32|counter |     blocks.follow(db)
32|counter |   File "/home/adam/counterparty-core/counterparty-core/counterpartycore/lib/blocks.py", line 1275, in follow
32|counter |     db.wal_checkpoint(mode=apsw.SQLITE_CHECKPOINT_PASSIVE)
32|counter | apsw.LockedError: LockedError: database table is locked
32|counter | 
32|counter | During handling of the above exception, another exception occurred:
32|counter | 
32|counter | Traceback (most recent call last):
32|counter |   File "/home/adam/.local/bin/counterparty-server", line 8, in <module>
32|counter |     sys.exit(cli.main())
32|counter |   File "/home/adam/counterparty-core/counterparty-core/counterpartycore/cli.py", line 445, in main
32|counter |     server.start_all(catch_up=args.catch_up)
32|counter |   File "/home/adam/counterparty-core/counterparty-core/counterpartycore/server.py", line 610, in start_all
32|counter |     db.close()
32|counter | apsw.ThreadingViolationError: You are trying to use the same object concurrently in two threads or re-entrantly within the same thread which is not allowed.
32|counter | [2024-04-20 11:18:06][DEBUG] Creating connection to `/home/adam/.local/share/counterparty/counterparty.db`.
32|counter | Traceback (most recent call last):
32|counter |   File "/home/adam/counterparty-core/counterparty-core/counterpartycore/server.py", line 598, in start_all
32|counter |     blocks.follow(db)
32|counter |   File "/home/adam/counterparty-core/counterparty-core/counterpartycore/lib/blocks.py", line 1275, in follow
32|counter |     db.wal_checkpoint(mode=apsw.SQLITE_CHECKPOINT_PASSIVE)
32|counter | apsw.LockedError: LockedError: database table is locked
32|counter | 
32|counter | During handling of the above exception, another exception occurred:
32|counter | 
32|counter | Traceback (most recent call last):
32|counter |   File "/home/adam/.local/bin/counterparty-server", line 8, in <module>
32|counter |     sys.exit(cli.main())
32|counter |   File "/home/adam/counterparty-core/counterparty-core/counterpartycore/cli.py", line 445, in main
32|counter |     server.start_all(catch_up=args.catch_up)
32|counter |   File "/home/adam/counterparty-core/counterparty-core/counterpartycore/server.py", line 610, in start_all
32|counter |     db.close()
32|counter | apsw.ThreadingViolationError: You are trying to use the same object concurrently in two threads or re-entrantly within the same thread which is not allowed.
```

**Comment by mattmarcello** [comment]
Does this happen consistently when you kill the process?  Also, just want to verify that you are seeing this on `develop`?

**Comment by adamkrellenstein** [comment]
no this was a random server crash that I saw!

**Comment by adamkrellenstein** [comment]
maybe sentry caught it??

**Comment by mattmarcello** [comment]
TLDR I think we can close this. 

I'm pretty sure I know what was going on here.  We did not know that an `apsw.Connection` was not thread safe and didn't notice for awhile because the api server was configured to be single threaded for a period of time.  My guess is that this happened when we configured the server to be threaded, but had not yet provided the telemetry daemon thread with it's own `apsw.Connection` instance ( which we are doing now in master  https://github.com/CounterpartyXCP/counterparty-core/commit/a5cb6ba1f3a616f0ab28385246955a2c292841fd ).
